kong/plugins/correlation-id/handler.lua

```diff
@@ -1,78 +1,91 @@
 -- Copyright (C) Kong Inc.
-
 local BasePlugin = require "kong.plugins.base_plugin"
-local req_set_header = ngx.req.set_header
-local req_get_headers = ngx.req.get_headers
-local uuid = require("kong.tools.utils").uuid
+local uuid = require "kong.tools.utils".uuid
 
-local CorrelationIdHandler = BasePlugin:extend()
 
-CorrelationIdHandler.PRIORITY = 1
-CorrelationIdHandler.VERSION = "0.1.0"
+local kong = kong
+
 
 local worker_uuid
 local worker_counter
+local generators
+
 
-local fmt = string.format
-local now = ngx.now
-local worker_pid = ngx.worker.pid()
-
-local generators = setmetatable({
-  ["uuid"] = function()
-    return uuid()
-  end,
-  ["uuid#counter"] = function()
-    worker_counter = worker_counter + 1
-    return worker_uuid .. "#" .. worker_counter
-  end,
-  ["tracker"] = function()
-    local var = ngx.var
-    return fmt("%s-%s-%d-%s-%s-%0.3f",
-      var.server_addr,
-      var.server_port,
-      worker_pid,
-      var.connection, -- connection serial number
-      var.connection_requests, -- current number of requests made through a connection
-      now() -- the current time stamp from the nginx cached time.
-    )
-  end,
-}, { __index = function(self, generator)
-    ngx.log(ngx.ERR, "Invalid generator: " .. generator)
+do
+  local worker_pid = ngx.worker.pid()
+  local now = ngx.now
+  local fmt = string.format
+
+  generators = setmetatable({
+    ["uuid"] = function()
+      return uuid()
+    end,
+    ["uuid#counter"] = function()
+      worker_counter = worker_counter + 1
+      return worker_uuid .. "#" .. worker_counter
+    end,
+    ["tracker"] = function()
+      local var = ngx.var
+      return fmt("%s-%s-%d-%s-%s-%0.3f",
+        var.server_addr,
+        var.server_port,
+        worker_pid,
+        var.connection, -- connection serial number
+        var.connection_requests, -- current number of requests made through a connection
+        now() -- the current time stamp from the nginx cached time.
+      )
+    end,
+  }, { __index = function(_, generator)
+      kong.log.err("Invalid generator: " .. generator)
```

Comment:
This extra metatable that's added for error-checking only is effectively dead code, because the schema guarantees that `conf.generator` is always valid. I think we could remove this and make `generators` a simple table.

Reply:
(I realize this was in the original code already, just noticed that and thought we could clean it up)